python/ray/train/batch_predictor.py

```diff
@@ -177,6 +192,15 @@ def __call__(self, batch):
         ray_remote_args["num_cpus"] = num_cpus_per_worker
         ray_remote_args["num_gpus"] = num_gpus_per_worker
 
+        if separate_gpu_stage and num_gpus_per_worker > 0:
+            preprocessor = self.get_preprocessor()
+            if preprocessor:
+                # Set the in-predictor preprocessing to a no-op when using a separate
+                # GPU stage. Otherwise, the preprocessing will be applied twice.
+                override_prep = BatchMapper(lambda x: x)
+                prep_fn = preprocessor.transform_batch
+                data = data.map_batches(prep_fn, batch_format="pandas")
```

Comment:
For consistency (and to support non-pandas Preprocessors)
```suggestion
                data = preprocessor.transform(data)
```